src/Controller/CRUDController.php

```diff
@@ -5,18 +5,78 @@
 
 namespace Zicht\Bundle\AdminBundle\Controller;
 
+use Doctrine\ORM\Mapping\ClassMetadata;
 use Sonata\AdminBundle\Controller\CRUDController as BaseCRUDController;
 use Symfony\Component\Form\FormRenderer;
 use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\PropertyAccess\PropertyAccess;
 use Symfony\Component\Security\Core\Exception\AccessDeniedException;
 use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Zicht\Bundle\AdminBundle\Event\AdminEvents;
+use Zicht\Bundle\AdminBundle\Event\ObjectDuplicateEvent;
+use Symfony\Component\PropertyInfo\Extractor\ReflectionExtractor;
+use Symfony\Component\PropertyInfo\PropertyInfoExtractor;
 
 /**
  * Provides some basic utility functionality for admin controllers, to be supplied as an construction parameter
  */
 class CRUDController extends BaseCRUDController
 {
+    /**
+     * Override content of original page with content from new page and remove new page
+     *
+     * @return RedirectResponse
+     */
+    public function overrideAction()
+    {
+        $id = $this->getRequest()->get($this->admin->getIdParameter());
+        $object = $this->admin->getObject($id);
+        $originalPage = $object->getCopiedFrom();
+
+        // Override properties without association mapping (like titel, introductie etc)
+        $propertyAccessor = PropertyAccess::createPropertyAccessor();
+
+        $fieldMappings = $this->getDoctrine()->getManager()->getClassMetadata(get_class($object))->getFieldNames();
+
+        foreach ($fieldMappings as $property) {
+            $value = $propertyAccessor->getValue($object, $property);
+            $propertyAccessor->setValue($originalPage, $property, $value);
+        }
+
+        // Override properties with association mapping/relations (like contentitems)
+        $associationMappings = $this->getDoctrine()->getManager()->getClassMetadata(get_class($object))->getAssociationNames();
+
+        foreach ($associationMappings as $property) {
+            if ($property === 'copiedFrom') {
+                continue;
+            }
+            $values = $propertyAccessor->getValue($object, $property);
+            $mappingType = $this->getDoctrine()->getManager()->getClassMetadata(get_class($object))->getAssociationMappings()[$property]['type'];
+
+            if ($mappingType === ClassMetadata::ONE_TO_MANY) {
+                $clonedValues = [];
+                foreach ($values as $value) {
+                    $clonedValues[] = clone $value;
+                }
+                $propertyAccessor->setValue($originalPage, $property, $clonedValues);
+            } else {
+                $propertyAccessor->setValue($originalPage, $property, $values);
+            }
+        }
+
+        $objectManager = $this->getDoctrine()->getManager();
+        $objectManager->remove($object);
+        $objectManager->flush();
```

Comment:
Ik zou het toch prettiger vinden om hier het oude object expliciet te persisten, ondanks dat we ervan mogen uitgaan dat de entity manager het object onder beheer heeft.